### PR TITLE
Add ruleset reference of PCRE2 attribute for expressions

### LINKED ISSUE #2907: Update the ruleset reference with the PCRE2 attribute for expressions
We are introducing a new parameter `type` in the `<rule>` and `<decoder>` options mentioned in:

- https://github.com/wazuh/wazuh/issues/5968
- https://github.com/wazuh/wazuh/issues/5970

On the other hand, we should add new content in:

- Custom rules and decoders
- Regular Expression Syntax

### PR DESCRIPTION
**Pull request by jnasselle**
Closes #2907 

## Description

Hi team!
This PR aims to add information regarding to PCRE2 regex type and support for the three(`osregex`, `osmatch` and `pcre2`) regex types to almost all fields and options related to regex. This consist in the next changes

Rules syntax: 
- [X] Affected options/fields
  - [X] regex
  - [X] field
  - [X] match
  - [X] action
  - [X] extra_data
  - [X] hostname
  - [X] id
  - [X] location
  - [X] match
  - [X] program_name
  - [X] protocol
  - [X] user
  - [X] url
  - [X] srcport
  - [X] dstport
  - [X] status
  - [X] system_name
  - [X] data
  - [X] srcgeoip
  - [X] dstgeoip
- [X] Tasks
  - [X] Update value to `Any regular expression` and description at index table for affected options/fields
  - [X] Add `type` as optional attribute for options/fields section, with their value range an default value
  - [X] Check that all references to regular expressions types are correct
  - [X] Add example(s)

Decoders syntax: 
- [X] Affected options/fields
  - [X] prematch
  - [X] regex
  - [X] program_name
- [X] Tasks
  - [X] Update value to `Any regular expression` and description at index table for affected options/fields
  - [X] Add `type` as optional attribute for options/fields section, with their value range an default value
  - [X] Check that all references to regular expressions types are correct
  - [X] Add example(s)

Regular expression syntax:
-  [X] Add PCRE2 heading with brief and link to their own documentation
-  [X] Add samples of new features bring by PCRE2 compared with osregex and osmatch

Other fixes:
-  [X] Rules: Fix links to `regex` option and Regular Expression Syntax page
-  [X] Rules: Add notes about regex concatenation
-  [x] Rules: Add information about attributes(`negate` and `type`) behavior in regex concatenation
-  [X] Rules: Arrange options information according to index
-  [X] Rules: `category` option: fix link to `type` (decoders), making reference to their value range.
-  [X] Rules: Change `name` to an attribute table (mandatory) in `field` option
-  [X] Decoders: `type` option: add to index, add table with all supported values, add ref for external links

## Checks
- [X] It compiles without warnings.
- [X] Spelling and grammar. 
- [X] Used impersonal speech. 
- [X] Used uppercase only on nouns. 
- [X] All links works as expected 

Regards,
Nico

### CODE REVIEW

File: source/user-manual/ruleset/ruleset-xml-syntax/decoders.rst [hunk 1]
```diff
@@ -88,7 +92,7 @@ There is many options to configure the decoders:
 - `accumulate`_
 - `program_name`_
 - `prematch`_
-- `regex`_
+- :ref:`regex_decoders`
```
Comment (davidfmiranda) [inline source/user-manual/ruleset/ruleset-xml-syntax/decoders.rst:95]:
```suggestion
- Regex`_

```

Comment (davidfmiranda) [reply source/user-manual/ruleset/ruleset-xml-syntax/decoders.rst:95]: We should always use this syntax in indices

**Review by davidfmiranda — CHANGES_REQUESTED** [review]
Hello @jnasselle, I've reviewed your PR and LGTM, nevertheless, there is a reference in an index. We must avoid this and use the syntax `Title`_.

Regards,

David

**Review by davidfmiranda — APPROVED** [review]
LGTM.

**Review by juliancnn — APPROVED** [review]

**Review by juliancnn — APPROVED** [review]

**Review by Lopuiz — APPROVED** [review]

**Review by juliancnn — APPROVED** [review]

**Review by JcabreraC — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Pull request merged**